### PR TITLE
System optimizations

### PR DESCRIPTION
**Pull request by odudex**
### Description
Problem: A damaged camera currently renders devices unusable.

Solution: This PR addresses the issue by allowing Krux to operate without a functioning camera and includes some performance optimizations.

Changes:

- Krux can now function without a camera, enabling key loading from storage or manual input, as well as signing transactions using an SD card.
- The camera is reinitialized only when necessary, reducing load time.

Other Optimizations
- Battery status reading and its display are now decoupled from the menu rendering process, improving menu responsiveness on devices equipped with a PMU.
- Translation refactor reduces total firmware size in ~5% and increases free RAM at boot in ~3%

### What is the purpose of this pull request? 

- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [x] Other

### CODE REVIEW

**Comment by odudex** [comment]
Translation refactor reduces total firmware size in ~5% and increases free RAM at boot in ~3%

**Comment by jdlcdl** [comment]
I've visually reviewed the code changes in 30 files here.

Much of this I've tested on Amigo (whose camera works fine), but none since this pr was readied last week, nor any of the changes since "translation optimizations".   Currently I'm having personal hw problems and cannot build for my amigo but I expect this to be resolved soon and will edit this message when done.

**Comment by odudex** [comment]
Thank you Jean! I also ended up adding new Sparrow's SD message signing feature here too, and this will require a few more tests to be written.
I hope you have your hw issues solved without too much hassle 🙏.

File: src/krux/pages/home_pages/sign_message_ui.py [hunk 1]
```diff
@@ -41,159 +43,223 @@
...
+            message_filename, data = self.load_file(prompt=False)
+            return (data, FORMAT_NONE, message_filename)
+        return (None, None, "")
+
+    def _is_valid_derivation_path(self, derivation_path):
+        """Checks if the derivation path is valid"""
+        try:
+            parts = derivation_path.split("/")
+            return parts[0] == "m" and all(
+                p.endswith("'") or p.endswith("h") or p.isdigit() for p in parts[1:]
```
Comment (jdlcdl) [inline src/krux/pages/home_pages/sign_message_ui.py:77]: note: check the tolower happens prior to this... or add "H"

File: src/krux/pages/home_pages/sign_message_ui.py [hunk 2]
```diff
@@ -41,159 +43,223 @@
...
-
-                    # Encode sig as base64 string
-                    encoded_sig = base_encode(sig, 64).strip().decode()
-                    self.ctx.display.clear()
-                    self.ctx.display.draw_centered_text(
-                        t("Signature") + ":\n\n%s" % encoded_sig
-                    )
-                    self.ctx.input.wait_for_button()
-                    return sig
+        script_from_deriv = int(parts[1].strip("'").strip("h"))
```
Comment (jdlcdl) [inline src/krux/pages/home_pages/sign_message_ui.py:100]: same as last comment

**Pull request merged**